### PR TITLE
[WIP] Optional billing

### PR DESCRIPTION
**Pull request by brunnoferreira**

### CODE REVIEW

File: Airwallex/Card/Internal/AWXCardViewModel.m [hunk 1]
```diff
@@ -24,31 +29,73 @@ - (instancetype)initWithSession:(AWXSession *)session {
...
+        return;
+    }
+    
+    AWXPlaceDetails *place = placeDetails.copy;
+    AWXAddress *address = addressDetails.copy;
+    place.address = address;
+    
+    *error = [place validate];
+    if (error == nil) {
+        self.validatedBilling = place;
```
Comment (aw-hector) [inline Airwallex/Card/Internal/AWXCardViewModel.m:57]: Nice work on creating new copy to submit request without potentially affecting the original reference passed from VC!

File: Airwallex/Card/Internal/AWXCardViewController.m [hunk 2]
```diff
@@ -134,6 +134,77 @@ - (void)viewDidLoad {
...
+    }
+
+    _confirmButton = [AWXActionButton new];
+    _confirmButton.enabled = YES;
+    [_confirmButton setTitle:NSLocalizedString(@"Confirm", @"Confirm") forState:UIControlStateNormal];
+    [_confirmButton addTarget:self action:@selector(confirmPayment:) forControlEvents:UIControlEventTouchUpInside];
+    [stackView addArrangedSubview:_confirmButton];
+    [_confirmButton.heightAnchor constraintEqualToConstant:52].active = YES;
+
+    if (self.session.billing) {
```
Comment (aw-hector) [inline Airwallex/Card/Internal/AWXCardViewController.m:148]: I'm even thinking to completely remove session from VC, all those view related display info will refer to the view model instead. But no essentially in this task.

File: Airwallex/Card/Internal/AWXCardViewModel.m (hunk 1 above)
Comment (brunnoferreira) [reply Airwallex/Card/Internal/AWXCardViewModel.m:57]: I know I said "not right now" but couldn't help myself haha. Made things more organised in the view model too.

File: Airwallex/Card/Internal/AWXCardViewController.m [hunk 3]
```diff
@@ -284,38 +299,39 @@ - (void)stopAnimating {
...
-    _firstNameField.hidden = sameAsShipping;
-    _lastNameField.hidden = sameAsShipping;
-    _countryView.hidden = sameAsShipping;
-    _stateField.hidden = sameAsShipping;
-    _cityField.hidden = sameAsShipping;
-    _streetField.hidden = sameAsShipping;
-    _zipcodeField.hidden = sameAsShipping;
-    _emailField.hidden = sameAsShipping;
-    _phoneNumberField.hidden = sameAsShipping;
+- (void)setBillingInputNotRequired:(BOOL)isNotRequired {
```
Comment (aw-hector) [inline Airwallex/Card/Internal/AWXCardViewController.m:302]: What about a more straightforward name `setBillingInputHidden:(BOOL)hidden` just for readability?

File: Airwallex/Card/Internal/AWXCardViewModel.m [hunk 4]
```diff
@@ -7,9 +7,13 @@
 //
 
 #import "AWXCardViewModel.h"
+#import "AWXSession.h"
+#import "AWXCard.h"
+#import "AWXCardProvider.h"
 
 @interface AWXCardViewModel ()
 
+@property (nonatomic, readwrite) BOOL isReusingShippingAsBillingInformation;
```
Comment (aw-hector) [inline Airwallex/Card/Internal/AWXCardViewModel.m:16]: Isn't a property `readwrite` by default?

**Pull request merged**